python/nav/ipdevpoll/snmp/common.py

```diff
@@ -134,10 +139,119 @@ def _getbulk(self, *args, **kwargs):
         return super(AgentProxyMixIn, self)._getbulk(*args, **kwargs)
 
 
-# pylint: disable=C0103
-SNMPParameters = namedtuple('SNMPParameters', 'timeout max_repetitions throttle_delay')
-
-SNMP_DEFAULTS = SNMPParameters(timeout=1.5, max_repetitions=50, throttle_delay=0)
+@dataclass
+class SNMPParameters:
+    """SNMP session parameters common to all SNMP protocol versions"""
+
+    # Common for all SNMP sessions
+    version: Union[Literal[1], Literal[2], Literal[3]] = 1
+    timeout: float = 1.5
+    tries: int = 3
+
+    # Common for v1 and v2 only
+    community: str = "public"
+
+    # Common for v2c +
+    max_repetitions: int = 50
+
+    # SNMPv3 only
+    sec_level: Optional[
+        Union[Literal["noAuthNoPriv"], Literal["authNoPriv"], Literal["authPriv"]]
```

Comment:
Actually, the tests are failing on Python 3.7 because I introduced the use of `Literal` here.

We don't actually use the type annotations for anything automated (like actual type checking), and we're not quite ready to drop Python 3.7 support yet, so I'm thinking maybe just replace this with `str`.  What do you think, @hmpf ?

Reply:
Or maybe better yet: Switch to using the existing Enums in `nav.Snmp,.defines`